lib/kolekti_metricfu/parsers/saikuro.rb

```diff
@@ -2,13 +2,19 @@ module KolektiMetricfu
   module Parsers
     class Saikuro < Base
       def self.parse(collected_metrics_hash, metric_configuration, persistence_strategy)
+
+
         collected_metrics_hash[:files].each do |file|
           name_prefix = parse_file_name(file[:filename])
-          file[:classes].each do |clazz|
-            clazz[:methods].each do |method|
+          file[:classes].each do |cls|
+            # Filter duplicate methods, picking only the latest. Fortunately Hash::[] will pick the latest element
+            # when receiveing an array.
+            methods = Hash[cls[:methods].map { |method| [method[:name], method] }]
```

Comment:
Can we assign the map result to a variable with a good name? This should improve code readability.